core/templates/dev/head/domain/collection/SkillListObjectFactorySpec.js

```diff
@@ -87,12 +89,41 @@ describe('Skill list object factory', function() {
     expect(skillList.getSkills()).toEqual(['skill2', 'skill0', 'skill1']);
   });
 
+  it('should be able to detect being a superset of another skill list',
+      function() {
+    var skillList1 = SkillListObjectFactory.create(['skill0', 'skill1']);
+    var skillList2 = SkillListObjectFactory.create(['skill0']);
+
+    expect(skillList1.isSupersetOfSkillList(skillList2)).toBeTruthy();
+    expect(skillList2.isSupersetOfSkillList(skillList1)).toBeFalsy();
+
+    // The empty list is a subset of all lists.
+    var skillList3 = SkillListObjectFactory.create([]);
+    expect(skillList1.isSupersetOfSkillList(skillList3)).toBeTruthy();
+    expect(skillList2.isSupersetOfSkillList(skillList3)).toBeTruthy();
+    expect(skillList3.isSupersetOfSkillList(skillList1)).toBeFalsy();
+    expect(skillList3.isSupersetOfSkillList(skillList2)).toBeFalsy();
+
+    // A list is the superset of itself, including the empty list.
+    expect(skillList1.isSupersetOfSkillList(skillList1)).toBeTruthy();
+    expect(skillList2.isSupersetOfSkillList(skillList2)).toBeTruthy();
+    expect(skillList3.isSupersetOfSkillList(skillList3)).toBeTruthy();
+
+    // A list can be made the superset of another.
```

Comment:
Not totally sure what this is testing. If it's a before/after thing, then maybe there should be an expect clause before the mutations are made to skillList4?